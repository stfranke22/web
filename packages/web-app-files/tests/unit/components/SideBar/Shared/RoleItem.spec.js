import stubs from '@/tests/unit/stubs/index.js'
import { createLocalVue, shallowMount } from '@vue/test-utils'
import RoleItem from '@files/src/components/SideBar/Shared/RoleItem'

const localVue = createLocalVue()

const filesPersonalRoute = { name: 'files-personal' }

function getWrapper(route, role) {
  return shallowMount(RoleItem, {
    localVue,
    stubs: stubs,
    mocks: {
      $route: route
    },
    propsData: {
      role
    }
  })
}

describe('RoleItem', () => {
  const role = {
    id: 'dfdd3eddde',
    label: 'Viewer',
    description: 'Download, Preview and Share'
  }
  const wrapper = getWrapper(filesPersonalRoute, role)

  it('should display the roleItem correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
