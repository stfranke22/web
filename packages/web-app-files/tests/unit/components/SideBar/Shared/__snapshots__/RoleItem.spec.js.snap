// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoleItem should display the roleItem correctly 1`] = `<span id="files-role-undefined" class="roles-select-role-item"><span class="oc-text-bold uk-display-block uk-width-1-1 roles-select-role-item-label">Viewer</span> <span class="oc-m-rm roles-select-role-item-description">Download, Preview and Share</span></span>`;
